### PR TITLE
Fix issues with cell outputs not being shown when doing Run All Cells in an Interactive notebook.

### PR DESCRIPTION
**Pull request by corivera**
When using Interactive notebooks with lots of cells, I noticed that some cells would succeed but the outputs wouldn't be updated. Looks like this is because there's a race condition between the runCell promise completing, and the cell outputs being manually updated from the Interactive extension. We only use a runCell Future promise with vscode notebooks for compatibility reasons, so it always returns undefined. So if that future promise completes after the cell outputs are updated from the extension, it sets the cell output to an empty array. My solution here to wait for any pending future promises to complete before updating the cell outputs.

### CODE REVIEW

File: src/sql/workbench/services/notebook/browser/models/cell.ts [hunk 1]
```diff
@@ -1098,6 +1098,15 @@ export class CellModel extends Disposable implements ICellModel {
 		this.fireOutputsChanged(false);
 	}
 
+	public processEdits(edits: ICellEdit[]): void {
+		// Wait for any in-progress operations to complete before doing any edits so that we don't hit any race conditions with updating cell outputs
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/notebook/browser/models/cell.ts:1102]: Isn't this going to break being able to update outputs as the cell is running though? That was specifically something I made sure could happen - the idea being that a cell may want to send partial/streamed results while it's executing.

Comment (corivera) [reply src/sql/workbench/services/notebook/browser/models/cell.ts:1102]: Do you have an example of code that would stream output like that?

Comment (Charles-Gagnon) [reply src/sql/workbench/services/notebook/browser/models/cell.ts:1102]: Not anymore - and this was for a custom test kernel I made when I was implementing this stuff. It hinges around using [appendOutput](https://github.com/microsoft/vscode/blob/main/src/vscode-dts/vscode.d.ts#L13277) on the Notebook extension APIs to modify the output of a cell during execution.

Comment (corivera) [reply src/sql/workbench/services/notebook/browser/models/cell.ts:1102]: I'll take a look and see how this change impacts that

Comment (corivera) [reply src/sql/workbench/services/notebook/browser/models/cell.ts:1102]: So the outputs still show up correctly when using append, but they all get displayed at once when the execution is done. I'll change this to fix that.

**Comment by corivera** [comment]
Disregard this PR. Looks like the underlying issue is that the Interactive extension doesn't await the promise for clearing the cell output before executing the code cell.

**Pull request closed without merging**